flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/windowing/WindowOperator.java

```diff
@@ -466,9 +466,12 @@ public void onEventTime(InternalTimer<K, W> timer) throws Exception {
             mergingWindows = getMergingWindowSet();
             W stateWindow = mergingWindows.getStateWindow(triggerContext.window);
             if (stateWindow == null) {
-                // Timer firing for non-existent window, this can only happen if a
-                // trigger did not clean up timers. We have already cleared the merging
-                // window and therefore the Trigger state, however, so nothing to do.
+                // Timer firing for non-existent window, this can happen:
```

Comment:
I actually find the old comment good enough and easier to understand than the updated version.

Reply:
That is embarrassing embarrassing :sweat_smile:. I tried to cover the false `ContinuousProcessingTimeTrigger.clear`. How about I keep old structure but adding a tailing node ? The final result should looks like:

> this can only happen if a trigger did not clean up timers(possibly due to state merged out). 

Or should I just revert ?

Reply:
There is nothing to be embarrassed about ;) I also always struggle to use English as it is not my primary language...

Isn't it already covered by: `We have already cleared the merging window and therefore the Trigger state` ?

Reply:
I guess there are different. The sentence you quoted describes more about window and trigger state clearing. Timers could be treated as part of trigger state depends on perspective. However, in merging case, these timers could be left over due to bookkeeping state merged out in `onMerge`, eg. there are not cleared. This is why we need the short-circuit return statement: the window firing timer belongs to is merged and cleared but the timer remains until firing.

It is my personal understanding. Anyway, I will revert it first. We could add a note back if required in later reviewing.